runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -2231,10 +2231,27 @@ J9::Z::TreeEvaluator::inlineCRC32CUpdateBytes(TR::Node *node, TR::CodeGenerator
    generateRREInstruction(cg, TR::InstOpCode::LLGFR, node, offset, offset);
    generateRREInstruction(cg, TR::InstOpCode::LLGFR, node, end, end);
 
+   // Offset to be added to array object pointer to get to the data elements
+   int32_t offsetToDataElements = isDirectBuffer ? 0 : TR::Compiler->om.contiguousArrayHeaderSizeInBytes();
+#ifdef J9VM_GC_SPARSE_HEAP_ALLOCATION
+   if (TR::Compiler->om.isOffHeapAllocationEnabled())
+      {
+      // Load first data element address
+      generateRXInstruction(cg,
+         TR::InstOpCode::getLoadOpCode(),
+         node,
+         array,
+         generateS390MemoryReference(array, cg->comp()->fej9()->getOffsetOfContiguousDataAddrField(), cg));
+
+      // Since the first data element address is retrieved from the array header, the offset is set to 0
+      offsetToDataElements = 0;
+      }
```

Comment:
Does this work for the `updateDirectByteBuffer` as well?

Reply:
I have just updated the if condition. From what I understand directByteBuffer is supposed to be a contiguous memory off heap so we should be good using the passed in address without doing anything special in off-heap mode. @amicic Can you please confirm?

Reply:
Not expert, but yes DBB probably has an explicit pointer to the memory.  DBB has nothing to do with Java Arrays, hence no dataAddr awareness (but I guess, one you reach the memory, you are free to apply some of array specific optimizations).